### DISCUSSION TITLE
AsusFX504 - 0.8.6

### DISCUSSION THREAD

**Discussion by RobyRew** [RobyRew/ASUS-FX504GE-Hackintosh_OpenCore#17] [General]
### News / Fixes:
- Updated my personal Laptop to Ventura 13.0
- Updated all config.plist parameters for OC 0.8.6, all OpenCore files and Kexts are also updated.
- Removed `SSDT-dGPU-Off.aml` because its already doing the same on `config.plist` > `DeviceProperties` > `PciRoot(0x0)/Pci(0x2,0x0)` > `disable-external-gpu`.
- Used [the new method](https://openintelwireless.github.io/IntelBluetoothFirmware/FAQ.html#what-additional-steps-should-i-do-to-make-bluetooth-work-on-macos-monterey-and-newer) of getting Bluetooth to work, should be better than before.
- Removed SUPPORT DEBUG versions of EFI.

### Be aware:
- Right now this EFI is using [Airportitlwm.kext v2.2.0-Alpha(ee56708)](https://github.com/OpenIntelWireless/itlwm/releases/tag/v2.2.0-alpha) because its using Alpha software it might have some issues unresolved, but nothing "Stable" is provided for Ventura yet.

### Known Issues:
- HDMI may not work great on certain displays especially if connected to a High Res Display.
- Audio sometimes or in some situations is not great.

_Pull Requests are welcome! :)_

<hr /><em>This discussion was created from the release <a href='https://github.com/RobyRew/ASUS-FX504GE-Hackintosh_OpenCore/releases/tag/0.8.6'>AsusFX504 - 0.8.6</a>.</em>

**Comment by RobyRew** [comment]
<!DOCTYPE html>

Realtek | ALC255/ALC3234 | layout 3, 11, 12, 13, 15, 17, 18, 20, 21, 22, 27, 28, 29, 30, 31, 66, 71, 82, 86, 96, 99, 100, 255 | 13 (10.9) | —
-- | -- | -- | -- | --

https://github.com/acidanthera/AppleALC/wiki/Supported-codecs

New layouts where added, testing needs to be done.

**Comment by code4power** [comment]
It does not work for me, load until the middle then stay and finally restart the laptop!

**Reply by RobyRew** [reply]
> It does not work for me, load until the middle then stay and finally restart the laptop!

I'm not sure about a fix, but What i'm thinking right now is to change the AirportItlwm.kext for the Monterrey version of stable 2.1.0
https://github.com/OpenIntelWireless/itlwm/releases/tag/v2.1.0

just try replacing the existing kext with the Monterrey version 2.1.0, no other modifications needed.

tell me if worked or if not.

cheers!

**Reply by code4power** [reply]
YES! That's exactly what I changed (AirportItlwm) and it works now , I say I use Macos Monterey.
(!) Boot very hard, two minutes !!! And I use M2 Samsung Evo Plus 970 2TB !!!

**Reply by RobyRew** [reply]
awesome man!!
I'm really glad you figured out to make it work!
Here it stays for anyone who might need help in the future!

I thought on my next release to place on the EFI all Airportitlwm versions renamed with "-osname" at the end of each one and setting a max and min kernel limit so each one loads on each specific version of macOS, and doing the same for the Bluetooth mess (4 different kexts for 2 macos versions)

What do you think, is worth trying?

**Reply by RobyRew** [reply]
> (!) Boot very hard, two minutes !!! And I use M2 Samsung Evo Plus 970 2TB !!!

I just updated yesterday to Ventura trough doing a clean install, and now the boot time is like 6 seconds aprox, before it was like a minute and something.

My hipótesis is that macOS update slows the boot time or messes something on the boot, because if i don't apply any updates the boot time doesn't change, but after applying some major update(12.X) it just does something IDK what.

**Reply by code4power** [reply]
I can't go to macOS Ventura, I work in music and video and there are no more compatible programs for now.

**Reply by code4power** [reply]
My opinion is from EFI, something confuses to boot, looking for something ....

**Reply by RobyRew** [reply]
> My opinion is from EFI, something confuses to boot, looking for something ....

what you could try is to use OnyX and maybe from its settings rebuild the structures, do some cleaning, I recommend taking a deep look on each setting and that tool can really help you, I've used in the past and in the present and is very very powerful.

https://www.titanium-software.fr/en/onyx.html

be sure to download the Monterrey one since each app is for each major os release

**Comment by code4power** [comment]
"AirportItlwm.kext" alpha for Monterey = 10Mega Download 
"AirportItlwm.kext" stabile for Monterey = 600Mega Download 
Thanks! RobyRew (Traiasca Oltenia :)) de aici esti, nu?

**Reply by RobyRew** [reply]
yeah, the alpha is just bad, I'm using the alpha on Ventura and I have a very unstable conection using HotSpot...

Sunt de la țara Jud. Botoșani, om de sat. 😜

**Comment by code4power** [comment]
I work in the musical field, can I rely on the headphone jack to play the music for a few hours? (I test now, but I want some opinions.)

**Reply by RobyRew** [reply]
for me it works very well for a long period of time with Sony WH-1000XM4 over the jack 3.5 audio quality is 10/10, I think it has 93.000Hz.

It plays Tidal HiFi Plus perfectly fine 😇

**Comment by code4power** [comment]
SIP ON & OFF from OpenCore:
NVRAM - 7C436110-AB2A-4BBB-A880-FE41995C9F82 - csr-active-config: 67000000
SIP OFF: 67000000
SIP ON: 00000000
(!) Reset nVRAM

**Reply by RobyRew** [reply]
what is this about for?

**Reply by code4power** [reply]
Some programs need to have access to system files (a kind of root), ex: TotalFinder (a very good plugin for Finder)
Terminal: csrutil status

**Comment by Dv1101** [comment]
is there ventura python mac recovery available? Cause switching from windows and Dortania's Guide isn't  updated or have to boot first monterey and Update?

**Reply by code4power** [reply]
I only have macOS in the house, for me it's simple. But I recommend you put Monterey. macOS Ventura is new too.

**Reply by RobyRew** [reply]
> is there ventura python mac recovery available? Cause switching from windows and Dortania's Guide isn't updated or have to boot first monterey and Update?

`python3 macrecovery.py -b Mac-7BA5B2D9E42DDD94 -m 00000000000000000 download`
this is for Ventura

**Reply by Dv1101** [reply]
> > is there ventura python mac recovery available? Cause switching from windows and Dortania's Guide isn't updated or have to boot first monterey and Update?
> 
> `python3 macrecovery.py -b Mac-7BA5B2D9E42DDD94 -m 00000000000000000 download`
> this is for Ventura 

Thanks 👍🏻

**Comment by code4power** [comment]
Someone can help me with a solution ....
When he enters "Sleep", my Bluetooth no longer works, he recovers after I restart.

**Reply by Dv1101** [reply]
Known issue, Unfortunately currently no fix available.

**Reply by code4power** [reply]
thanks. Let's hope it will be resolved in the end ...

**Reply by RobyRew** [reply]
> thanks. Let's hope it will be resolved in the end ...

Try this:

Create a file named .wakeup on your home directory.
Paste this on it:

```
#!/bin/bash
#
# Restart Bluetooth Module on Mac OS X
#
# Requires Blueutil to be installed: http://brewformulas.org/blueutil

BT="/usr/local/bin/blueutil"

log() {
	echo "$@"
	logger -p notice -t bt_restarter "$@"
}

err() {
	echo "$@" >&2
	logger -p error -t bt_restarter "$@"
}

if [ -f "$BT" ]; then
	if [[ $("$BT" -p) == *1 ]];
		then
		echo "Bluetooth on, restarting ..."
		($("$BT" -p 0) &> /dev/null && echo "Bluetooth Module stopped") || (err "Couldn't stop Bluetooth Module" && exit 1)
		($("$BT" -p 1) &> /dev/null && echo "Bluetooth Module started") || (err "Couldn't start Bluetooth Module" && exit 1) 
		log "Successfully restarted Bluetooth" && exit 0
	else
		echo "Bluetooth is off, nothing to do ..."
	fi
else
	err "Couldn't find blueutil, please install http://brewformulas.org/blueutil" && exit 1
fi
```

The script makes use of BlueUtil, so you have to install it too. (I'm assuming you have[ HomeBrew](https://brew.sh/index) installed)
Open Terminal and run:

```bash
brew install sleepwatcher
brew install blueutil
```

To make use of SleepWatcher, you have to put the gist to ~/.wakeup (For more information about sleepwatcher run: man sleepwatcher)

After you put your script to ~/.wakeup do the following:

```bash
chmod +x ~/.wakeup
```

Now you can test the script. Add the following command to your terminal:
```bash
/usr/local/sbin/sleepwatcher --verbose -w ~/.wakeup
```

Now you have to enter sleep mode and wakeup your Mac again. After that, you should see the following console output:

```
Bluetooth on, restarting ...
Bluetooth Module stopped
Bluetooth Module started
Successfully restarted Bluetooth
sleepwatcher: wakeup: /Users/ralph/.wakeup: 0
```

If it works, you can set up launchd to start sleepwatcher at login by using brew services.

```bash
brew services start sleepwatcher
```

Credits: @ralph089 [(Link)](https://gist.github.com/ralph089/a65840c4f5e439b90170d735a89a863f)

**Comment by code4power** [comment]
Finally we have passed all the stages successfully but it does not work ... the service is started: homebrew.mxcl.sleepwatcher.plist

**Reply by RobyRew** [reply]
im gonna look more in the future

**Reply by code4power** [reply]
today I updated to 12.6.2., and my bluetooth works when it goes to sleep... I don't understand anything anymore.... :D

**Reply by RobyRew** [reply]
> today I updated to 12.6.2., and my bluetooth works when it goes to sleep... I don't understand anything anymore.... :D

incredible, wow

**Comment by Dv1101** [comment]
Hi, Does this will bring display scaling option? https://github.com/bbhardin/A-Guide-to-MacOS-Scaled-Resolutions
Currently there's no scaling option available. on 1080p things too much small. I tried tinkering tool as you suggested but still many things stays too much small. Only way to increase size of elements is at the cost of lower resolutions.

https://www.reddit.com/r/hackintosh/comments/see2rq/updated_to_122_and_can_no_longer_change_screen/

**Reply by RobyRew** [reply]
honestly I don't know anything about that, I'm very happy without scaling, I would want the items to be even smaller, since I need more things on the sreen

**Comment by Happsz1** [comment]
Eyo, btw a quick question, is GPU acceleration already enabled here by default in this premade config or nah?

**Reply by RobyRew** [reply]
fully working man!
the only things that are not working are indicated on the README.md

**Reply by Happsz1** [reply]
Thankies for letting me know! 

Was just asking cuz of me noticing that in programs like Blender and Photoshop are slower than I remember on my previous installs as my previous installs doesn't have issues with those programs in regard to performance (because yes I'm an artist that happily uses your Hackintosh config, albeit modified slightly by me lol)

(And yep I keep reinstalling my OS so many timesbcuz I had a hard time choicing which OS to use lol)

**Reply by Happsz1** [reply]
I also got a question too that is somewhat related to the GPU, and that question is: Any ways to fix the HDMI flickering? Is it a framebuffer issue or something? 

I enabled HDMI so I can use my drawing tablet (the HUION GT191), however there would be flickering sometimes, and sometimes there would be no flickering at all when I restarted my laptop.

**Reply by code4power** [reply]
macOS Monterey with OpenCore 0.9.1
Yes, the video card, default and HDMI works, but this you have to configure your drivers for NVIDIA. EFI that you find here is old and does not work the NVIDIA video card, plus this EFI has other defects on the side of Sleep ...

**Reply by RobyRew** [reply]
> I also got a question too that is somewhat related to the GPU, and that question is: Any ways to fix the HDMI flickering? Is it a framebuffer issue or something?

I has something to do with audio over HDMI, I don't know what exactly is but I've tried hundreds of different settings, the problem totally disappear when you disable audio completely form EFI, and there for HDMI won't get any audio, so the problem is either with the framebuffer on the iGPU, or Apple ALC audio, a post related of this issue (apparently) is is found on this link:
https://www.tonymacx86.com/threads/uhd-630-no-hdmi-audio.265490/page-5
But with my limited knowledge I don't understand that and I haven't figured out to get working.
 
> I enabled HDMI so I can use my drawing tablet (the HUION GT191), however there would be flickering sometimes, and sometimes there would be no flickering at all when I restarted my laptop.

As temporary you can disable your audio going to: `DeviceProperties` > `Add` > `PciRoot(0x0)/Pci(0x1F,0x3)` and move the entire tree to `Delete` key
something like this:
![image](https://user-images.githubusercontent.com/49403384/233835805-1ea3a262-9f02-4644-bf3b-c77a8d166f77.png)

PS: any fix you found or suggestion about the EFI please share so we can all improve our experience. 😉

**Reply by RobyRew** [reply]
> macOS Monterey with OpenCore 0.9.1
> Yes, the video card, default and HDMI works, but this you have to configure your drivers for NVIDIA. EFI that you find here is old and does not work the NVIDIA video card, plus this EFI has other defects on the side of Sleep ...

sadly NVIDIA card won't work no matter of what you do, there aren't web drivers for NVIDIA GPUs since macOS Hight Sierra, and this laptop beside that, has something called [Nvidia Optimus](https://www.nvidia.com/en-us/geforce/technologies/optimus/technology/) which switches between GPUs when load requires, that thing is disabled because any MacBook has that, and the only GPU routed directly to the screen and HDMI is the intel iGPU, which would work as a passthrough for the NVIDIA if NVIDIA Optimus wouldn't be there.

This laptop only uses de Intel UHD 630 as iGPU, the NVIDIA 1050Ti is disabled at boot.

**Reply by Happsz1** [reply]
> > I also got a question too that is somewhat related to the GPU, and that question is: Any ways to fix the HDMI flickering? Is it a framebuffer issue or something?
> 
> I has something to do with audio over HDMI, I don't know what exactly is but I've tried hundreds of different settings, the problem totally disappear when you disable audio completely form EFI, and there for HDMI won't get any audio, so the problem is either with the framebuffer on the iGPU, or Apple ALC audio, a post related of this issue (apparently) is is found on this link: https://www.tonymacx86.com/threads/uhd-630-no-hdmi-audio.265490/page-5 But with my limited knowledge I don't understand that and I haven't figured out to get working.
> 
> > I enabled HDMI so I can use my drawing tablet (the HUION GT191), however there would be flickering sometimes, and sometimes there would be no flickering at all when I restarted my laptop.
> 
> As temporary you can disable your audio going to: `DeviceProperties` > `Add` > `PciRoot(0x0)/Pci(0x1F,0x3)` and move the entire tree to `Delete` key something like this: ![image](https://user-images.githubusercontent.com/49403384/233835805-1ea3a262-9f02-4644-bf3b-c77a8d166f77.png)
> 
> PS: any fix you found or suggestion about the EFI please share so we can all improve our experience. 😉



I'ma try disabling HDMI audio to see if the issue does disappear and it is practical in my case to disable it as my drawing tablet doesn't even have speakers to begin with lol

**Reply by RobyRew** [reply]
great reply back if fixes the issue or not

**Reply by Happsz1** [reply]
Finally back here lol, I haven't tried disabling it via OpenCore Configurator or any other apps that can change the plist file, however, do you think that disabling the AppleGFXHDA kext would solve the HDMI audio corrupting the drawing tablet's display?

I did my research on disabling HDMI audio through different ways and heard that the AppleGFXHDA kext is responsible for handling audio for external displays, at least according to my Google searches on how to turn off HDMI audio.
